File: app/src/main/java/com/test/shoppingapp/shoppinglist/adapter/ShoppingListViewHolder.java
package com.test.shoppingapp.shoppinglist.adapter;

import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import com.test.networkmodule.response.ShopListingResponse;
import com.test.shoppingapp.R;

class ShoppingListViewHolder extends RecyclerView.ViewHolder {
    public final View mView;

    public final ImageView imageProduct;
    public final TextView numLikes;
    public final TextView numComments;
    public final TextView price;
    public final ImageView status;
    public final TextView name;

    public ShopListingResponse mItem;

    public ShoppingListViewHolder(View view) {
        super(view);
        mView = view;
        imageProduct = view.findViewById(R.id.img_product_pic);
        status = view.findViewById(R.id.img_status);
        numLikes = view.findViewById(R.id.num_likes);
        numComments = view.findViewById(R.id.num_comments);
        price = view.findViewById(R.id.price);
        name = view.findViewById(R.id.name);
    }
}

File: app/src/main/java/com/test/shoppingapp/shoppinglist/adapter/ShoppingListViewHolder.kt
package com.test.shoppingapp.shoppinglist.adapter

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.bumptech.glide.Glide
import com.test.networkmodule.response.ShopListingResponse
import com.test.shoppingapp.R
import com.test.shoppingapp.shoppinglist.ProductStatus

class ShoppingListViewHolder(val view: View) : RecyclerView.ViewHolder(view) {
    private val imageProduct: ImageView = view.findViewById(R.id.img_product_pic)
    private val numLikes: TextView = view.findViewById(R.id.num_likes)
    private val numComments: TextView = view.findViewById(R.id.num_comments)
    private val price: TextView = view.findViewById(R.id.price)
    private val status: ImageView = view.findViewById(R.id.img_status)
    private val name: TextView = view.findViewById(R.id.name)

    fun bind(shopListingResponse: ShopListingResponse) {
        name.text = shopListingResponse.name
        numLikes.text = shopListingResponse.numOfLikes.toString()
        numComments.text = shopListingResponse.numOfComments.toString()
        price.text = String.format("$%s", shopListingResponse.price.toString())

        if (ProductStatus.SOLD_OUT == shopListingResponse.status) {
            status.visibility = View.VISIBLE
        } else {
            status.visibility = View.GONE
        }

        Glide.with(imageProduct)
            .load(shopListingResponse.photo)
            .into(imageProduct)
    }
}